scope3_methodology/ad_tech_platform/model.py

```diff
@@ -0,0 +1,330 @@
+""" Model for computing emissions for an ad tech platform"""
+
+from utils.utils import log_result
+
+
+class ModeledAdTechPlatform:
+    """This represents a modeled node in our graph"""
+
+    def __init__(
+        self,
+        name: str,
+        identifier: str,
+        primary_bid_request_emissions_g_co2e: float,
+        primary_cookie_sync_emissions_g_co2e: float,
+        cookie_sync_distribution_rate: float,
+        bid_request_rejection_rate: float,
+        secondary_bid_request_emissions_g_co2e: float | None = None,
+        secondary_cookie_sync_emissions_g_co2e: float | None = None,
+    ) -> None:
+        self.name = name
+        self.identifier = identifier
+        self.primary_bid_request_emissions_g_co2e = primary_bid_request_emissions_g_co2e
+        self.primary_cookie_sync_emissions_g_co2e = primary_cookie_sync_emissions_g_co2e
+        self.cookie_sync_distribution_rate = cookie_sync_distribution_rate
+        self.bid_request_rejection_rate = bid_request_rejection_rate
+        self.secondary_bid_request_emissions_g_co2e = secondary_bid_request_emissions_g_co2e
+        self.secondary_cookie_sync_emissions_g_co2e = secondary_cookie_sync_emissions_g_co2e
+
+
+class DistributionPartner:
+    """This represents an edge in our graph"""
+
+    def __init__(
+        self,
+        partner: ModeledAdTechPlatform,
+        bid_request_distribution_rate: float,
+    ) -> None:
+        self.partner = partner
+        self.bid_request_distribution_rate = bid_request_distribution_rate
+
+
+class AdTechPlatform:
+    """Raw emissions information about an ATP and methodology of how to calculate emissions"""
+
+    def __init__(
```

Comment:
I'd consider making this a dataclass to get the standard __init__ method for free: https://docs.python.org/3/library/dataclasses.html